### ISSUE TITLE
Convert Hitsbe class to nn.Module and register buffers for device compatibility

### ISSUE THREAD

**Issue by angelolmn** [angelolmn/HitsBE#10]
The Hitsbe class must be updated to inherit from nn.Module and register important tensors (the attention mask and the positional embedding matrix) as buffers.
